libnmstate/nm/wired.py

```diff
@@ -0,0 +1,96 @@
+#
+# Copyright 2018 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+from libnmstate.ethtool import minimal_ethtool
+from libnmstate.nm import nmclient
+
+
+def create_setting(iface_state, base_con_profile):
+    mtu = iface_state.get('mtu')
+
+    ethernet = iface_state.get('ethernet', {})
+    speed = ethernet.get('speed')
+    duplex = ethernet.get('duplex')
+    auto_negotiation = ethernet.get('auto-negotiation')
+
+    if not (mtu or speed or duplex or (auto_negotiation is not None)):
+        return None
+
+    wired_setting = None
+    if base_con_profile:
+        wired_setting = base_con_profile.get_setting_wired()
+        if wired_setting:
+            wired_setting = wired_setting.duplicate()
+
+    if not wired_setting:
+        wired_setting = nmclient.NM.SettingWired.new()
+
+    if mtu:
+        wired_setting.set_property(nmclient.NM.SETTING_WIRED_MTU, mtu)
+
+    if speed:
+        wired_setting.set_property(nmclient.NM.SETTING_WIRED_SPEED, speed)
+
+    if duplex:
+        wired_setting.set_property(nmclient.NM.SETTING_WIRED_DUPLEX, duplex)
+
+    if auto_negotiation is not None:
+        wired_setting.set_property(nmclient.NM.SETTING_WIRED_AUTO_NEGOTIATE,
+                                   auto_negotiation)
+
+    return wired_setting
+
+
+def get_info(device):
+    """
+    Provides the current active values for a device
+    """
+    info = {}
+
+    iface = device.get_iface()
+    try:
+        info['mtu'] = int(device.get_mtu())
+    except AttributeError:
+        pass
+
+    ethernet = {}
+    try:
+        ethernet['speed'] = int(device.get_speed())
+    # DeviceWifi, DeviceGeneric, DeviceTun do not have a get_speed() method
```

Comment:
Somehow the comment here is gone.
I think it is clear without this comment.

Reply:
As a sign of good faith I removed it. :-)